backend/libraries/ballerina-core-go/sum.go

```diff
@@ -1,52 +1,22 @@
 package ballerina
 
-import (
-	"fmt"
-)
-
-type Sum[a any, b any] interface{}
-type left[a any, b any] struct {
-	Sum[a, b]
-	value a
+type Sum[a any, b any] struct {
+	IsLeft     bool
+	ValueLeft  a
+	ValueRight b
```

Comment:
There is a correct implementation in document-api. This is not isomorphic to Sum (as soon as ValueLeft and ValueRight are public this breaks)